numpy/ma/core.py

```diff
@@ -7683,6 +7676,10 @@ def asanyarray(a, dtype=None):
     <class 'numpy.ma.core.MaskedArray'>
 
     """
+    # workaround for #8666, to preserve identity. Ideally the bottom line
```

Comment:
I think one would add right after the [first line](https://github.com/numpy/numpy/blob/master/numpy/ma/core.py#L2766):
```
if _data is data and (type(_data) is cls or subok and isinstance(_data, cls)):
    return _data
```